fdbclient/MonitorLeader.actor.cpp

```diff
@@ -880,7 +881,7 @@ ACTOR Future<MonitorLeaderInfo> monitorProxiesOneGeneration(
 			bool upToDate = wait(connRecord->upToDate(storedConnectionString));
 			if (upToDate) {
 				incorrectTime = Optional<double>();
-			} else if (allConnectionsFailed) {
+			} else if (allConnectionsFailed && storedConnectionString.getNumberOfCoordinators() > 0) {
```

Comment:
How do we get a connection string with no coordinators? If this is related to us creating a `DatabaseContext` with an error and not initializing the other pieces, it might be better to avoid calling this function at all.

Reply:
If upToDate(storedConnectionString) fails to parse the new contents of the cluster file, it returns false and does not change storedConnectionString, so it stays as it was initialized by the default constructor of ClusterConnectionString, so to say empty.

Reply:
I reproduced crash by a test that starts with a cluster file with valid content and then corrupting its contents.